internal/pkg/service/stream/storage/level/local/encoding/pipeline.go

```diff
@@ -281,9 +281,13 @@ func (p *pipeline) WriteRecord(record recordctx.Context) error {
 	// Increments number of high-level writes in progress
 	p.acceptedWrites.Add(timestamp, 1)
 
+	// Set sync timeout
+	ctx, cancel := context.WithTimeout(record.Ctx(), 20*time.Second)
+	defer cancel()
+
 	// Wait for sync and return sync error, if any
-	if err := notifier.Wait(record.Ctx()); err != nil {
-		return err
+	if err := notifier.Wait(ctx); err != nil {
+		return errors.PrefixError(err, "error when waiting for sync")
 	}
```

Comment:
@Matovidlo I added here a 20s timeout, to see, if the `408` errors are caused by the long sync, or not.